api/provider/aws/common.go

```diff
@@ -243,3 +243,47 @@ func listClusterTaskARNs(ecsapi ecsiface.ECSAPI, clusterName, startedBy, status
 
 	return taskARNs, nil
 }
+
+func listContainerInstances(ecsapi ecsiface.ECSAPI, clusterName string) ([]string, error) {
+	containerInstanceARNs := []string{}
+	fn := func(output *ecs.ListContainerInstancesOutput, lastPage bool) bool {
+		for _, c := range output.ContainerInstanceArns {
+			containerInstanceARNs = append(containerInstanceARNs, aws.StringValue(c))
+		}
+
+		return !lastPage
+	}
+
+	input := &ecs.ListContainerInstancesInput{}
+	input.SetCluster(clusterName)
+
+	if err := ecsapi.ListContainerInstancesPages(input, fn); err != nil {
+		return nil, err
+	}
+
+	return containerInstanceARNs, nil
+}
+
+func listClusterServiceNames(ecsapi ecsiface.ECSAPI, clusterNames []string) ([]string, error) {
```

Comment:
don't we already have a helper that does this?

Reply:
Not that i'm aware of. That was I think one of the biggest issues with the existing code where it looked like we were getting services from all environments and then filter for a particular environment. I'm sure this would have been a major if not the biggest contributor to the rate limits we were seeing.

Reply:
Also same thing applies - this code is no longer used.